### ISSUE TITLE
templates / variables in `choices` 

### ISSUE THREAD

**Issue by exitcode0** [copier-org/copier#1359]
### Describe the problem

I'm new to both copier and jinja so perhaps I just need to be pointed in the correct direction

I'm trying to construct the `choices` option on a question

### Template

```
question_choices:
    default: ["A","B"]
    when: false

actual_question:
    help: "the cake is a lie"
    choices: ["{{question_choices}}"]
```

```
question_choices:
    default: ["A","B"]
    when: false

actual_question:
    help: "the cake is a lie"
    choices: "{% for item in {{question_choices}} -%}  - 'name': '{{item}}'{% endfor %}"
```

### Expected behavior

hoping to be able to generate the valid choices at run time, perhaps via script

### Operating system

macOS

### Operating system distribution and version

Ventura 13.6

### Copier version

copier 8.3.0

### Python version

Python 3.11.5

### Installation method

pip+git

**Comment by sisp** [comment]
I'm afraid what you'd like to do is currently not possible. See also #1193 for a previous discussion.

**Comment by sisp** [comment]
How about allowing a string value for `choices` which gets rendered, parsed as YAML, and validated? This way, we could achieve dynamic choice items without the need for a generic solution. I think it should be fairly straightforward to implement it.

**Comment by sisp** [comment]
This feature was implemented in #1804 and will become available in the next release.
